### PR TITLE
Des 414 edit a lesson within a topic

### PR DESCRIPTION
**Pull request by Daniielmayo**

### CODE REVIEW

File: components/EditLessonWithinATopic/EditLessonWithinATopic.module.css [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+  background: none;
+  border: none;
+  font-size: 1.5rem;
+  cursor: pointer;
+}
+
+.search-bar {
+  width: 100%;
+  padding: 0.75rem;
+  border: 1px solid var(--gray);
```
Comment (DanielCalleCO) [inline components/EditLessonWithinATopic/EditLessonWithinATopic.module.css:150]: px

File: components/EditLessonWithinATopic/EditLessonWithinATopic.module.css [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+
+.resource-list {
+  max-height: 20rem;
+  overflow-y: auto;
+}
+
+.resource-item {
+  padding: 0.75rem 1rem;
+  cursor: pointer;
+  border-bottom: 1px solid var(--lightGray);
```
Comment (DanielCalleCO) [inline components/EditLessonWithinATopic/EditLessonWithinATopic.module.css:163]: px

File: components/organisms/CreateLessonWithinCourseAdminPortal/CreateLessonWithinCourseAdminPortal.tsx [hunk 3]
```diff
@@ -27,7 +28,7 @@ export const CreateLessonWithinCourseAdminPortal = () => {
   const [selectedType, setSelectedType] = useState("tipo de recurso");
   const courseId = useAppSelector((state) => state.courses.selectedCourse?._id);
   const topicId = useAppSelector((state) => state.topics.selectedTopic?._id);
-
+
```
Comment (DanielCalleCO) [inline components/organisms/CreateLessonWithinCourseAdminPortal/CreateLessonWithinCourseAdminPortal.tsx:31]: delete this blank space please (not the line)

File: components/organisms/CreateLessonWithinCourseAdminPortal/CreateLessonWithinCourseAdminPortal.tsx [hunk 4]
```diff
@@ -74,7 +75,7 @@ export const CreateLessonWithinCourseAdminPortal = () => {
     const resourceId = selectedResourceData?._id;
     Swal.fire({
       title: "¿Estás seguro de agregar el recurso a este tema?",
-      text: "Vas a agregar el recurso a este tema",
+      text: "Se agregara el recurso a este tema",
```
Comment (DanielCalleCO) [inline components/organisms/CreateLessonWithinCourseAdminPortal/CreateLessonWithinCourseAdminPortal.tsx:78]:
```suggestion
      text: "Se agregaá el recurso a este tema.",
```

File: components/organisms/CreateLessonWithinCourseAdminPortal/CreateLessonWithinCourseAdminPortal.tsx [hunk 5]
```diff
@@ -175,7 +192,7 @@ export const CreateLessonWithinCourseAdminPortal = () => {
           <Button
             type="submit"
             className={styles["button__submit"]}
-            disabled={!selectedResource}
+            // disabled={!selectedResource}
```
Comment (DanielCalleCO) [inline components/organisms/CreateLessonWithinCourseAdminPortal/CreateLessonWithinCourseAdminPortal.tsx:195]: do we need this commented line?

File: components/organisms/CreateLessonWithinCourseAdminPortal/CreateLessonWithinCourseAdminPortal.tsx [hunk 6]
```diff
@@ -184,8 +201,9 @@ export const CreateLessonWithinCourseAdminPortal = () => {
         <hr className={styles["button__split"]} />
         <div className={styles["button__content-delete-course"]}>
           <Button
-            className={styles["button__delete-course"]}
-            disabled={!selectedResource}
+            className={styles["button__submit"]}
+            // disabled={!selectedResource}
```
Comment (DanielCalleCO) [inline components/organisms/CreateLessonWithinCourseAdminPortal/CreateLessonWithinCourseAdminPortal.tsx:205]: do we need this commented line?

File: components/organisms/LessonsWithinACourseAdminPortal/LessonsWithinACourseAdminPortal.tsx [hunk 7]
```diff
@@ -30,7 +32,22 @@ export const LessonsWithinACourseAdminPortal = () => {
...
+  const handleEditClick = (row: any) => {
+    if (row) {
+      dispatch(resourceEditAdmin(row));
+      router.push(
+        "/dashboard/adminPanel/courses/courseTopicManage/lessonsWithinACourse/editLesson"
+      );
+    } else {
+      Swal.fire({
+        title: "Error al seleccionar el recurso",
+        text: "Hablar con el equipo de desarrollo",
```
Comment (DanielCalleCO) [inline components/organisms/LessonsWithinACourseAdminPortal/LessonsWithinACourseAdminPortal.tsx:44]: I think this line should be more friendly... something like:

```suggestion
        text: "Ponte en contacto con el equipo de desarrollo",
```

File: components/organisms/CreateLessonWithinCourseAdminPortal/CreateLessonWithinCourseAdminPortal.tsx [hunk 8]
```diff
@@ -114,6 +115,22 @@ export const CreateLessonWithinCourseAdminPortal = () => {
...
+  const handleEditClick = () => {
+    if (selectedResourceData) {
+      router.push(
+        `/dashboard/adminPanel/resources/editResource/${selectedResourceData._id}`
+      );
+      dispatch(resourceEditAdmin(selectedResourceData));
+    } else {
+      Swal.fire({
+        title: "Seleccionar recurso para poder editar",
+        text: "No hay ningun recurso seleccionado",
```
Comment (DanielCalleCO) [inline components/organisms/CreateLessonWithinCourseAdminPortal/CreateLessonWithinCourseAdminPortal.tsx:127]:
```suggestion
        text: "No has seleccionado ningún recurso.",
```

File: components/organisms/CreateLessonWithinCourseAdminPortal/CreateLessonWithinCourseAdminPortal.tsx [hunk 9]
```diff
@@ -114,6 +115,22 @@ export const CreateLessonWithinCourseAdminPortal = () => {
...
   };
+  const handleEditClick = () => {
+    if (selectedResourceData) {
+      router.push(
+        `/dashboard/adminPanel/resources/editResource/${selectedResourceData._id}`
+      );
+      dispatch(resourceEditAdmin(selectedResourceData));
+    } else {
+      Swal.fire({
+        title: "Seleccionar recurso para poder editar",
```
Comment (DanielCalleCO) [inline components/organisms/CreateLessonWithinCourseAdminPortal/CreateLessonWithinCourseAdminPortal.tsx:126]:
```suggestion
        title: "Selecciona un recurso para editar",
```

**Review by DanielCalleCO — CHANGES_REQUESTED** [review]

File: components/organisms/CreateLessonWithinCourseAdminPortal/CreateLessonWithinCourseAdminPortal.tsx (hunk 6 above)
Comment (Daniielmayo) [reply components/organisms/CreateLessonWithinCourseAdminPortal/CreateLessonWithinCourseAdminPortal.tsx:205]: That line is not necessary in the code, it was removed.

File: components/organisms/CreateLessonWithinCourseAdminPortal/CreateLessonWithinCourseAdminPortal.tsx (hunk 5 above)
Comment (Daniielmayo) [reply components/organisms/CreateLessonWithinCourseAdminPortal/CreateLessonWithinCourseAdminPortal.tsx:195]: That line is not necessary in the code, it was removed.

File: components/EditLessonWithinATopic/EditLessonWithinATopic.module.css (hunk 1 above)
Comment (Daniielmayo) [reply components/EditLessonWithinATopic/EditLessonWithinATopic.module.css:150]: measures changed to rem

File: components/organisms/CreateLessonWithinCourseAdminPortal/CreateLessonWithinCourseAdminPortal.tsx (hunk 4 above)
Comment (Daniielmayo) [reply components/organisms/CreateLessonWithinCourseAdminPortal/CreateLessonWithinCourseAdminPortal.tsx:78]: image

File: components/organisms/LessonsWithinACourseAdminPortal/LessonsWithinACourseAdminPortal.tsx (hunk 7 above)
Comment (Daniielmayo) [reply components/organisms/LessonsWithinACourseAdminPortal/LessonsWithinACourseAdminPortal.tsx:44]: image

File: components/organisms/CreateLessonWithinCourseAdminPortal/CreateLessonWithinCourseAdminPortal.tsx (hunk 8 above)
Comment (Daniielmayo) [reply components/organisms/CreateLessonWithinCourseAdminPortal/CreateLessonWithinCourseAdminPortal.tsx:127]: image

**Review by DanielCalleCO — APPROVED** [review]

**Pull request merged**